src/zarr/storage/_zep8.py

```diff
@@ -0,0 +1,661 @@
+        else:
+            # No colon - treat entire spec as adapter name
+            adapter = spec
+            path = ""
+
+        return URLSegment(adapter=adapter, path=path)
+
+    def resolve_relative(self, base: URLSegment, relative_path: str) -> URLSegment:
+        """
+        Resolve a relative path against a base URLSegment.
+
+        Parameters
+        ----------
+        base : URLSegment
+            Base URL segment to resolve against.
+        relative_path : str
+            Relative path to resolve.
+
+        Returns
+        -------
+        URLSegment
+            New URLSegment with resolved path.
+        """
+        if not relative_path:
+            return base
+
+        if relative_path.startswith("/"):
+            # Absolute path - replace base path
+            return URLSegment(scheme=base.scheme, adapter=base.adapter, path=relative_path)
+
+        # Relative path - combine with base path
+        base_path = base.path
+        if base_path and not base_path.endswith("/"):
+            base_path += "/"
+
+        new_path = base_path + relative_path
+        return URLSegment(scheme=base.scheme, adapter=base.adapter, path=new_path)
+
+    @staticmethod
+    def resolve_relative_url(base_url: str, relative_url: str) -> str:
+        """
+        Resolve relative URLs using Unix-style .. syntax.
+
+        Currently only supports Unix-style relative paths (e.g., "../path/file.zarr").
+        Pipe-prefixed relative URLs (e.g., "|..|file.zarr") are not implemented.
+
+        Parameters
+        ----------
+        base_url : str
+            The base ZEP 8 URL to resolve against.
+        relative_url : str
+            Unix-style relative URL with .. components.
+
+        Returns
+        -------
+        str
+            The resolved URL (currently just returns relative_url unchanged).
+
+        Examples
+        --------
+        >>> URLParser.resolve_relative_url(
+        ...     "s3://bucket/data/exp1.zip|zip:|zarr3:",
+        ...     "../control.zip|zip:|zarr3:"
+        ... )
+        '../control.zip|zip:|zarr3:'
+        """
+        # Currently only supports Unix-style relative paths by returning them unchanged
+        # TODO: Implement proper relative URL resolution if needed
+        return relative_url
+
+
+def is_zep8_url(url: Any) -> bool:
+    """
+    Check if a string is a ZEP 8 URL.
+
+    ZEP 8 URLs are identified by:
+    1. Presence of pipe (|) characters (for chained URLs)
+    2. Simple adapter syntax like "memory:", "zip:", etc. (single segment)
+
+    Parameters
+    ----------
+    url : str
+        String to check.
+
+    Returns
+    -------
+    bool
+        True if the string appears to be a ZEP 8 URL.
+
+    Examples
+    --------
+    >>> is_zep8_url("s3://bucket/data.zip|zip:|zarr3:")
+    True
+    >>> is_zep8_url("memory:")
+    True
+    >>> is_zep8_url("s3://bucket/data.zarr")
+    False
+    >>> is_zep8_url("file:///data.zarr")
+    False
```

Comment:
I don't really follow these returning false. seems like a downgrade in functionality, also having looked a the spec a bit I don't see where this is explicitly disallowed (though very possible i misread or misunderstood)